messaging/socketmaster.cc

```diff
@@ -51,26 +51,29 @@ struct SubMaster::SubMessage {
   bool updated = false, alive = false, valid = true, ignore_alive;
   uint64_t rcv_time = 0, rcv_frame = 0;
   void *allocated_msg_reader = nullptr;
+  bool is_polled = false;
   capnp::FlatArrayMessageReader *msg_reader = nullptr;
   AlignedBuffer aligned_buf;
   cereal::Event::Reader event;
 };
 
-SubMaster::SubMaster(const std::vector<const char *> &service_list, const char *address,
-                     const std::vector<const char *> &ignore_alive) {
+SubMaster::SubMaster(const std::vector<const char *> &service_list, const std::vector<const char *> &poll,
+                     const char *address, const std::vector<const char *> &ignore_alive) {
   poller_ = Poller::create();
   for (auto name : service_list) {
     const service *serv = get_service(name);
     assert(serv != nullptr);
     SubSocket *socket = SubSocket::create(message_context.context(), name, address ? address : "127.0.0.1", true);
     assert(socket != 0);
-    poller_->registerSocket(socket);
+    bool is_polled = inList(poll, name);
```

Comment:
default of empty vector should mean poll on all
```suggestion
    bool is_polled = poll.empty() || inList(poll, name);
```